modules/etcd-creating-automated-backups.adoc

```diff
@@ -0,0 +1,320 @@
++
+[WARNING]
+====
+Deleting or otherwise losing access to the node containing the stored backup data can cause unrecoverable data loss.
+====
+
+.. Create a `StorageClass` CR file named `etcd-backup-local-storage.yaml` with the following contents:
++
+[source,yaml]
+----
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: etcd-backup-local-storage
+provisioner: kubernetes.io/no-provisioner
+volumeBindingMode: WaitForFirstConsumer
+----
+
+.. Apply the `StorageClass` CR by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f etcd-backup-local-storage.yaml
+----
+
+.. Create a PV named `etcd-backup-pv-fs.yaml` from the applied `StorageClass` with content such as the following:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: etcd-backup-pv-fs
+spec:
+  capacity:
+    storage: 100Gi <.>
+  volumeMode: Filesystem
+  accessModes:
+  - ReadWriteMany
+  persistentVolumeReclaimPolicy: Delete
+  storageClassName: local-storage
+  local:
+    path: /mnt/
+  nodeAffinity:
+    required:
+      nodeSelectorTerms:
+      - matchExpressions:
+        - key: kubernetes.io/hostname
+          operator: In
+          values:
+          - <example-master-node> <.>
+----
+<.> The amount of storage available to the PV. Adjust this value for your requirements.
+<.> Replace this value with the node to attach this PV to.
++
+[TIP]
+====
+Run the following command to list the available nodes:
+
+[source,terminal]
+----
+$ oc get nodes
+----
+====
+
+.. Verify the creation of the PV by running the following command:
++
+[source,terminal]
+----
+$ oc get pv
+----
++
+.Example output
+[source,terminal]
+----
+NAME                    CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM   STORAGECLASS    REASON   AGE
+etcd-backup-pv-fs       100Gi      RWX            Delete           Available           local-storage            10s
+----
+
+.. Create a PVC named `etcd-backup-pvc.yaml` with contents such as the following:
++
+[source,yaml]
+----
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: etcd-backup-pvc
+spec:
+  accessModes:
+  - ReadWriteMany
+  volumeMode: Filesystem
+  resources:
+    requests:
+      storage: 10Gi <.>
+  storageClassName: local-storage
+----
+<.> The amount of storage available to the PVC. Adjust this value for your requirements.
+
+.. Apply the PVC by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f etcd-backup-pvc.yaml
+----
+
+. Create a custom resource definition (CRD) file named `etcd-recurring-backups.yaml`. The contents of the created CRD define the schedule and retention type of automated backups.
++
+For the default retention type, use contents such as the following:
++
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1alpha1
+kind: Backup
+metadata:
+  name: etcd-recurring-backup
+spec:
+  etcd:
+    schedule: "20 4 * * *" <.>
+    timeZone: "UTC"
+    retentionPolicy:
+      retentionType: "" <.>
```

Comment:
quickly testing this yields:

```
[tjungblu ~]$ kubectl apply -f /tmp/test.yaml 
The Backup "etcd-recurring-backup" is invalid: 
* spec.etcd.retentionPolicy.retentionType: Unsupported value: "": supported values: "RetentionNumber", "RetentionSize"
* <nil>: Invalid value: "": "spec.etcd.retentionPolicy.retentionType" must validate all the schemas (allOf)
```

are you sure that the default "" works? :)

Reply:
Hi @kowen-rh ,
Please refer this https://docs.google.com/document/d/1lWKE0689aChScmqeFvCxuMHRZD0M-Gv-GlscAgpjwAs/edit#heading=h.dds4j7gs8mu3 where this test scenario has been captured and executed with successful logs. 

Below is the yaml file for validating the default value of retentiontype.

apiVersion: config.openshift.io/v1alpha1
kind: Backup
metadata:
   name: testbackup-h
spec:
   etcd:
      schedule: "*/3 * * * *"
      timeZone: "UTC"
      pvcName: <corresponding pvc-name>

Reply:
Are you suggesting that the literal value `""` itself doesn't work, but simply not specifying a value does?

Reply:
yes.